csharp/src/Microsoft.ML.OnnxRuntime/OrtAllocator.cs

```diff
@@ -27,6 +27,53 @@ public enum OrtMemType
         Default = 0,                        // the default allocator for execution provider
     }
 
+    /// <summary>
+    /// This class encapsulates arena configuration information that will be used to define the behavior
+    /// of an arena based allocator
+    /// See ONNX_Runtime_Perf_Tuning.md for more details
+    /// </summary>
+    public class OrtArenaCfg : SafeHandle
+    {
+        /// <summary>
+        /// Create an instance of arena configuration which will be used to create an arena based allocator
+        /// See docs/C_API.md for details on what the following parameters mean and how to choose these values
+        /// </summary>
+        /// <param name="maxMemory">Maximum amount of memory the arena allocates</param>
+        /// <param name="arenaExtendStrategy">Strategy for arena expansion</param>
+        /// <param name="initialChunkSizeBytes">Size of the region that the arena allocates first</param>
+        /// <param name="maxDeadBytesPerChunk">Maximum amount of fragmentation allowed per chunk</param>
+        /// <returns>An instance of OrtArenaCfg</returns>
+        public OrtArenaCfg(uint maxMemory, int arenaExtendStrategy, int initialChunkSizeBytes, int maxDeadBytesPerChunk)
+            : base(IntPtr.Zero, true)
+        {
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtCreateArenaCfg((UIntPtr)maxMemory,
+                                                                           arenaExtendStrategy,
+                                                                           initialChunkSizeBytes,
+                                                                           maxDeadBytesPerChunk,
+                                                                           out handle));
+        }
+
+        public override bool IsInvalid { get { return handle == IntPtr.Zero; } }
+
+        internal IntPtr Pointer
+        {
+            get
+            {
+                return handle;
+            }
+        }
+
+        #region SafeHandle
+        protected override bool ReleaseHandle()
+        {
```

Comment:
Document it

Reply:
Example for both [https://github.com/microsoft/onnxruntime/pull/5853/files#diff-8dfe8f0895dc274d1cd472df892bbe7e67ae0ed66dc1f0f1c444c953f20b10d5L136](https://github.com/microsoft/onnxruntime/pull/5853/files#diff-8dfe8f0895dc274d1cd472df892bbe7e67ae0ed66dc1f0f1c444c953f20b10d5L136)

---
In reply to: [527234407](https://github.com/microsoft/onnxruntime/pull/5634#discussion_r527234407) [](ancestors = 527234407)

Reply:
Done